alerting/src/main/kotlin/org/opensearch/alerting/util/ClusterMetricsCoordinator.kt

```diff
@@ -0,0 +1,386 @@
+//            val failureReasons3 = checkShardsFailure(indexResponse3)
+//            val failureReasons4 = checkShardsFailure(indexResponse4)
+//            val failureReasons5 = checkShardsFailure(indexResponse5)
+//            val failureReasons6 = checkShardsFailure(indexResponse6)
+//            val failureReasons7 = checkShardsFailure(indexResponse7)
+//            val failureReasons8 = checkShardsFailure(indexResponse8)
+//            val failureReasons9 = checkShardsFailure(indexResponse9)
+//            if (
+//                failureReasons != null ||
+//                failureReasons2 != null ||
+//                failureReasons3 != null ||
+//                failureReasons4 != null ||
+//                failureReasons5 != null ||
+//                failureReasons6 != null ||
+//                failureReasons7 != null ||
+//                failureReasons8 != null ||
+//                failureReasons9 != null
+//            ) {
+//                log.info("failed because $failureReasons")
+//                log.info("failed because $failureReasons2")
+//                log.info("failed because $failureReasons3")
+//                log.info("failed because $failureReasons4")
+//                log.info("failed because $failureReasons5")
+//                log.info("failed because $failureReasons6")
+//                log.info("failed because $failureReasons7")
+//                log.info("failed because $failureReasons8")
+//                log.info("failed because $failureReasons9")
+//                return
+//            }
+//        } catch (t: Exception) {
+//            log.info("Unable to get index response,  $t")
+//        }
+    }
+    private fun checkShardsFailure(response: IndexResponse): String? {
+        val failureReasons = StringBuilder()
+        if (response.shardInfo.failed > 0) {
+            response.shardInfo.failures.forEach {
+                    entry ->
+                failureReasons.append(entry.reason())
+            }
+            return failureReasons.toString()
+        }
+        return null
+    }
+
+    private fun deleteDocs(client: NodeClient) {
+        val documentAge = metricsStoreTime.toString()
+
+        ClusterMetricsDataPoint.MetricType.values().forEach {
+            DeleteByQueryRequestBuilder(client, DeleteByQueryAction.INSTANCE)
+                .source(ClusterMetricsVisualizationIndex.CLUSTER_METRIC_VISUALIZATION_INDEX)
+                .filter(QueryBuilders.rangeQuery(it.metricName + ".timestamp").lte("now-$documentAge"))
+                .execute(
+                    object : ActionListener<BulkByScrollResponse> {
+                        override fun onResponse(response: BulkByScrollResponse) {
+                        }
+
+                        override fun onFailure(t: Exception) {
+                        }
+                    }
+                )
+            log.info("deleted ${it.metricName} data from $documentAge ago")
```

Comment:
Question for other reviewers: Do we think it would be worthwhile to include the index name in this log message? For example:
```
log.info("Deleted ${it.metricName} data from $documentAge ago from ${ClusterMetricsVisualizationIndex.CLUSTER_METRIC_VISUALIZATION_INDEX}.")